packages/react-native/ReactAndroid/src/test/java/com/facebook/react/modules/deviceinfo/DeviceInfoModuleTest.kt

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+package com.facebook.react.modules.deviceinfo
+
+import com.facebook.react.bridge.Arguments
+import com.facebook.react.bridge.JavaOnlyMap
+import com.facebook.react.bridge.ReactApplicationContext
+import com.facebook.react.bridge.ReactContext
+import com.facebook.react.bridge.ReactTestHelper
+import com.facebook.react.bridge.WritableMap
+import com.facebook.react.uimanager.DisplayMetricsHolder
+import junit.framework.TestCase
+import org.assertj.core.api.Assertions
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.ArgumentCaptor
+import org.mockito.ArgumentMatchers
+import org.mockito.Mockito.*
+import org.powermock.api.mockito.PowerMockito
+import org.powermock.api.mockito.PowerMockito.`when` as whenever
+import org.powermock.core.classloader.annotations.PowerMockIgnore
+import org.powermock.core.classloader.annotations.PrepareForTest
+import org.powermock.modules.junit4.rule.PowerMockRule
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.RuntimeEnvironment
+
+@RunWith(RobolectricTestRunner::class)
+@PrepareForTest(Arguments::class, DisplayMetricsHolder::class)
+@PowerMockIgnore("org.mockito.*", "org.robolectric.*", "androidx.*", "android.*")
+class DeviceInfoModuleTest : TestCase() {
+  @get:Rule var rule = PowerMockRule()
+
+  private lateinit var deviceInfoModule: DeviceInfoModule
+  private lateinit var fakePortraitDisplayMetrics: WritableMap
+  private lateinit var fakeLandscapeDisplayMetrics: WritableMap
+  private lateinit var reactContext: ReactApplicationContext
+
+  @Before
+  public override fun setUp() {
+    initTestData()
+    PowerMockito.mockStatic(DisplayMetricsHolder::class.java)
+    reactContext = spy(ReactApplicationContext(RuntimeEnvironment.application))
+    val catalystInstanceMock = ReactTestHelper.createMockCatalystInstance()
+    reactContext.initializeWithInstance(catalystInstanceMock)
+    deviceInfoModule = DeviceInfoModule(reactContext)
+  }
+
+  @After
+  fun teardown() {
+    DisplayMetricsHolder.setWindowDisplayMetrics(null)
+    DisplayMetricsHolder.setScreenDisplayMetrics(null)
+  }
+
+  @Test
+  fun test_itDoesNotEmitAnEvent_whenDisplayMetricsNotChanged() {
+    givenDisplayMetricsHolderContains(fakePortraitDisplayMetrics)
+    deviceInfoModule.typedExportedConstants
```

Comment:
Is this supposed to have side effects?.. Looks confusing.

Reply:
Guess it was just there to check that the getter doesn't crash, or something...

Reply:
Yup that's the reason behind this